### ISSUE TITLE
Mongo Repositories support for Java 8 Instant?

### ISSUE THREAD

**Issue by StephenOTT** [immutables/immutables#887]
seeing a error related to bson and Java 8 Instant 

when creating a mongo repo that has a <Instant> attribute, get the folllowing

`org.bson.codecs.configuration.CodecConfigurationException: Can't find a codec for class java.time.Instant.`

**Comment by elucash** [comment]
Yep, we don't have all the mappings. But you can provide any bindings you need, I believe. I'm not sure if you use Jackson/or Gson based binding mechanism, you could try either `CodecRegistry` on `RepositorySetup` or redefining `Gson` instance with type adapters. If you don't care about the particular type, you can try to use `org.immutables.mongo.types.TimeInstant`, which might work out of the box.

Sorry, with recent refactoring to use CodecRegistry and use Jackson bindings in addition to Gson, while this allowed for more integration scenarios, I'm less sure what exactly can work in your setup. With Gson, you can implement adapter factory akin to `org.immutables.mongo.types.TypeAdapters` and register it using GsonBuilder see `RepositorySetup.Builder.gson` javadoc (or via classpath using `@Metainf.Service` annotation on the custom `TypeAdapterFactory` implementation).

**Comment by asereda-gs** [comment]
Immutables doesn't provide serialization for `java.time` out of box (due to jdk7 support).

However, I think Mongo driver supports it natively since 3.7 / 3.8 (read [what's new](http://mongodb.github.io/mongo-java-driver/3.8/whats-new/)).

As Eugene pointed out, depending on your setup you may or may not to register custom Gson TypeAdapter or Jackson (de-)serializer for JSR-310.

Pls note that Jackson support is currently experimental.

**Comment by StephenOTT** [comment]
Ya i ended up finding the unit test that setup the jackson mappings... makes life WAYYYYY eaiser.  Would be amazing if this was documented in the website docs as the usage is really not obvious..

my curent usage is something like

```
        MongoDatabase database = new MongoClient("localhost", 27017).getDatabase("someDb");

        TenantDocumentRepository TENANTS = new TenantDocumentRepository(
                RepositorySetup.builder()
                        .codecRegistry(JacksonCodecs.registryFromMapper(StixParsers.getJsonMapper(true)))
                        .executor(MoreExecutors.newDirectExecutorService()).database(database).build());

```

**Comment by asereda-gs** [comment]
Agree about documentation. We're not yet strongly publicizing jackson support in immutables.

**Comment by StephenOTT** [comment]
Is there any thoughts on having a proper spring integration (or other major framework) so the custom repos are not required?  I know there is discussion of the "generic repo" support, but generally it would seem the issue is a pattern or steps to follow to support immutables ?

**Comment by asereda-gs** [comment]
What exactly do you mean by "spring integration" ? 

Technically you can just use [JacksonCodecs](https://github.com/immutables/immutables/blob/master/mongo/src/org/immutables/mongo/bson4jackson/JacksonCodecs.java) with raw mongo driver to save your objects.

**Comment by elucash** [comment]
`*dreaming*` we need to live in a world where no "integration" is necessary, there's just a tool and you just use it )

**Comment by StephenOTT** [comment]
>However, I think Mongo driver supports it natively since 3.7 / 3.8 (read what's new).

Ya i was just looking at 3.8.  Any issues with upgrading?

**Comment by asereda-gs** [comment]
We have some test dependencies ([Fongo](https://github.com/fakemongo/fongo). Possible one needs to migrate to [mongo-java-server](https://github.com/bwaldvogel/mongo-java-server)) which don't work with 3.7+ but you can enforce different version of mongo driver for your project. 

See #831
